src/main/java/hudson/plugins/build_timeout/impl/ElasticTimeOutStrategy.java

```diff
@@ -68,9 +81,17 @@ public long getTimeOut(AbstractBuild<?, ?> build, BuildListener listener)
             throws InterruptedException, MacroEvaluationException, IOException {
         double elasticTimeout = getElasticTimeout(Integer.parseInt(expandAll(build,listener,getTimeoutPercentage())), build, listener);
         if (elasticTimeout == 0) {
-            return Math.max(MINIMUM_TIMEOUT_MILLISECONDS, Integer.parseInt(expandAll(build, listener, getTimeoutMinutesElasticDefault())) * MINUTES);
+            if (isFailSafeTimeoutDuration()) {
+                return Math.max(Integer.parseInt(getTimeoutMinutesElasticDefault()) * 60L * 1000L, Integer.parseInt(expandAll(build, listener, getTimeoutMinutesElasticDefault())) * MINUTES);
+            } else {
+                return Math.max(MINIMUM_TIMEOUT_MILLISECONDS, Integer.parseInt(expandAll(build, listener, getTimeoutMinutesElasticDefault())) * MINUTES);
+            }
         } else {
-            return (long) Math.max(MINIMUM_TIMEOUT_MILLISECONDS, elasticTimeout);
+            if (isFailSafeTimeoutDuration()) {
+                return Math.max(Integer.parseInt(getTimeoutMinutesElasticDefault()) * 60L * 1000L, (long) elasticTimeout);
```

Comment:
* We need handle the case `timeoutMinutesElasticDefault` contains variable expressions. Let's use `expandAll`.
* In the case `timeoutMinutesElasticDefault` is set to `0` (or the value less than `MINIMUM_TIMEOUT_MILLISECONDS`, the return value is expected to be `MINIMUM_TIMEOUT_MILLISECONDS`.